presto-main/src/main/java/com/facebook/presto/metadata/FunctionManager.java

```diff
@@ -306,6 +306,15 @@ private FunctionHandle resolveFunctionInternal(Optional<TransactionId> transacti
             return new BuiltInFunctionHandle(getMagicLiteralFunctionSignature(type));
         }
 
+        Optional<SqlFunction> udtFunction = builtInFunctionNamespaceManager.getUdtFunctionImplementation(
```

Comment:
This logic should be in `BuiltInFunctionNamespaceManager`. `FunctionManager` should delegate function resolution to the `FunctionNamespaceManager` managing the specific function namespace. There should be no special handling necessary.

Reply:
went for a different approach using type constraints so I can have static function signatures here